Source/FieldSolver/SpectralSolver/SpectralAlgorithms/AvgGalileanAlgorithm.cpp

```diff
@@ -0,0 +1,371 @@
+                S3(i,j,k) = std::sin(1.5_rt*c*k_norm*dt);
+
+                // Calculate dot product with galilean velocity
+                const Real kv = modified_kx[i]*vx +
+#if (AMREX_SPACEDIM==3)
+                                 modified_ky[j]*vy +
+                                 modified_kz[k]*vz;
+#else
+                                 modified_kz[j]*vz;
+#endif
+
+                const Real nu = kv/(k_norm*c);
+                const Complex theta = amrex::exp( 0.5_rt*I*kv*dt );
+                const Complex theta_star = amrex::exp( -0.5_rt*I*kv*dt );
+                const Complex e_theta = amrex::exp( I*c*k_norm*dt );
+
+                Theta2(i,j,k) = theta*theta;
+
+                if ( (nu != 1.) && (nu != 0) ) {
+
+                    // Note: the coefficients X1, X2, X3 do not correspond
+                    // exactly to the original Galilean paper, but the
+                    // update equation have been modified accordingly so that
+                    // the expressions/ below (with the update equations)
+                    // are mathematically equivalent to those of the paper.
+                    Complex x1 = 1._rt/(1._rt-nu*nu) *
+                        (theta_star - C(i,j,k)*theta + I*kv*S_ck(i,j,k)*theta);
+
+                    Complex C_rho = I* c2 /( (1._rt-theta*theta) * ep0);
+
+                    Psi1(i,j,k) = theta * ((S1(i,j,k) + I*nu*C1(i,j,k))
+                                  - Theta2(i,j,k) * (S3(i,j,k) + I*nu*C3(i,j,k))) /(c*k_norm*dt * (nu*nu - 1._rt));
+                    Psi2(i,j,k) = theta * ((C1(i,j,k) - I*nu*S1(i,j,k))
+                                  - Theta2(i,j,k) * (C3(i,j,k) - I*nu*S3(i,j,k))) /(c2*k_norm*k_norm*dt * (nu*nu - 1._rt));
+                    Psi3(i,j,k) = I * theta * (1._rt - theta*theta) /(c*k_norm*dt*nu);
+
+                    A1(i,j,k) = (Psi1(i,j,k)  - 1._rt + I * kv*Psi2(i,j,k)    )/ (c2* k_norm*k_norm * (nu*nu - 1._rt));
+                    A2(i,j,k) = (Psi3(i,j,k) - Psi1(i,j,k)) / (c2*k_norm*k_norm);
+
+                    CRhoold(i,j,k) = C_rho * (theta*theta * A1(i,j,k) - A2(i,j,k));
+                    CRhonew(i,j,k) = C_rho * (A2(i,j,k) - A1(i,j,k));
+                    Jcoef(i,j,k) = (I*kv*A1(i,j,k) + Psi2(i,j,k))/ep0;
+                    // x1, above, is identical to the original paper
+                    X1(i,j,k) = theta*x1/(ep0*c*c*k_norm*k_norm);
+                    // The difference betwen X2 and X3 below, and those
+                    // from the original paper is the factor ep0*k_norm*k_norm
+                    X2(i,j,k) = (x1 - theta*(1._rt - C(i,j,k)))
+                                /(theta_star-theta)/(ep0*k_norm*k_norm);
+                    X3(i,j,k) = (x1 - theta_star*(1._rt - C(i,j,k)))
+                                /(theta_star-theta)/(ep0*k_norm*k_norm);
+                    X4(i,j,k) = I*kv*X1(i,j,k) - theta*theta*S_ck(i,j,k)/ep0;
+                }
+                if ( nu == 0) {
```

Comment:
Here I think it would be better to add the suffix `_rt` for single precision runs:
```suggestion
                if ( nu == 0._rt) {
```